### PR TITLE
dummy provider and threading bugfix

### PR DESCRIPTION
**Pull request by piotrm0**

### CODE REVIEW

File: trulens_eval/examples/dummy_example.ipynb [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# hugs = Huggingface()\n",
+    "hugs = Dummy()\n",
+    "\n",
+    "f_positive_sentiment = Feedback(hugs.positive_sentiment).on_output()"
```
Comment (rshih32) [inline trulens_eval/examples/dummy_example.ipynb:68]: How does the dummy have a positive_sentiment attribute?

Comment (piotrm0) [reply trulens_eval/examples/dummy_example.ipynb:68]: It has the same feedback functions as Huggingface except they always produce the same result, without making any actual huggingface requests.

File: trulens_eval/examples/dummy_example.ipynb [hunk 2]
```diff
@@ -0,0 +1,126 @@
+{
```
Comment (joshreini1) [inline trulens_eval/examples/dummy_example.ipynb:1]: <p>Please remove this dev cell</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/truera/trulens/pull/418/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (joshreini1) [inline trulens_eval/examples/dummy_example.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #18.        force = True,
</pre><p>Please remove dev params</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/truera/trulens/pull/418/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (joshreini1) [inline trulens_eval/examples/dummy_example.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #4.            print(f"thread count={threading.active_count()}, promises={TP().promises.qsize()}", end="")
</pre><p>Can you add a quick markdown cell explaining what you're doing here? Specifically the promises?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/truera/trulens/pull/418/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

**Review by rshih32 — APPROVED** [review]

**Pull request merged**